qiskit/circuit/controlledgate.py

```diff
@@ -60,3 +60,14 @@ def __eq__(self, other):
             return (other.num_ctrl_qubits == self.num_ctrl_qubits and
                     self.base_gate == other.base_gate and
                     super().__eq__(other))
+
+    def inverse(self):
+        """Invert this gate by calling inverse on the base gate."""
+        params = self.params
+        if self.base_gate is Gate:
```

Comment:
Can you add a test which visits this?

Reply:
perhaps test_inverse_circuit already does this sufficiently.

Reply:
Yes this is how I noticed handling "pure Gates" was necessary, due to test_inverse_circuit failing.